manifest.go

```diff
@@ -567,6 +569,96 @@ func ReadManifestList(in io.Reader) ([]ManifestFile, error) {
 	return out, dec.Error()
 }
 
+// WriteManifestListV2 writes a list of v2 manifest files to an avro file.
+func WriteManifestListV2(out io.Writer, files []ManifestFile) error {
+	return writeManifestList(out, files, 2)
+}
+
+// WriteManifestListV1 writes a list of v1 manifest files to an avro file.
+func WriteManifestListV1(out io.Writer, files []ManifestFile) error {
+	return writeManifestList(out, files, 1)
+}
+
+func writeManifestList(out io.Writer, files []ManifestFile, version int) error {
+	for _, file := range files {
+		if file.Version() != version {
+			return ErrInvalidArgument
+		}
+	}
+
+	var key string
+	switch version {
+	case 1:
+		key = internal.ManifestListV1Key
+	case 2:
+		key = internal.ManifestListV2Key
+	default:
+		return ErrInvalidArgument
+	}
+
+	enc, err := ocf.NewEncoder(
+		internal.AvroSchemaCache.Get(key).String(),
```

Comment:
use `NewEncoderWithSchema(internal.AvroSchemaCache.Get(key), out, ...)` and skip the conversion to string

do we need to manually add `avro.schema` key? or will that be added automatically by the `ocf` encoder?

Reply:
Hmm, I thought we needed the schemas here because of the field-ids, but I have very little avro experience, so I'll defer to you.

Reply:
Scracht that, I misunderstood your question - I'll check on the `avro.schema` key.

Reply:
Hmm, so it does not look like using `avro.NewEncoderWithSchema` works here (nor does is accept the codec/format-version options or, from what I understand, the ability to add metadata. It's not clear to me what to do here.

Reply:
Why doesn't it work? it accepts the same list of `EncoderFunc` options, so the arguments should work identically just removing the `.String()` from the first argument.

What happens that fails if you try using `ocf.NewEncoderWithSchema`?

Reply:
I see the issue - v2.26.0 doesn't have the func you're referring to. I'll bump to v2.27.0

Reply:
and to answer your other question, `avro.schema` gets added automatically. https://github.com/hamba/avro/blob/20ac18bf16cb5642333dc1441370a39501cc68de/ocf/ocf.go#L346

Looks like codec does as well - should I remove that?

Reply:
yea, if they get added automatically we can remove the manual setting of them.